### PR TITLE
fix: consensus invalid block and version test improvement

### PR DESCRIPTION
**Pull request by singhhp1069**
* fix https://github.com/gnolang/gno/issues/2773
* minor improvement and testcases for tm2/pkg/versionset
<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: tm2/pkg/bft/state/validation_test.go [hunk 1]
```diff
@@ -36,40 +36,55 @@ func TestValidateBlockHeader(t *testing.T) {
...
 		/*
 			Invalid blocks don't pass
 		*/
 		for _, tc := range testCases {
 			block, _ := state.MakeBlock(height, makeTxs(height), lastCommit, proposerAddr)
 			tc.malleateBlock(block)
 			err := blockExec.ValidateBlock(state, block)
-			require.Error(t, err, tc.name)
+
+			require.Error(t, err, "%s", tc.name)
```
Comment (thehowl) [inline tm2/pkg/bft/state/validation_test.go:83]: why change?

**Review by thehowl — COMMENTED** [review]
I need a second review to finish reviewing the VersionSet code.

But in general, it's not necessary that these are joined together (they don't depend on one another); please put the version test improvements in another PR>

File: tm2/pkg/bft/state/validation_test.go (hunk 1 above)
Comment (singhhp1069) [reply tm2/pkg/bft/state/validation_test.go:83]: my bad

**Comment by thehowl** [comment]
https://github.com/gnolang/gno/pull/2859 was merged, which is more complete

**Pull request closed without merging**